core.coffee

```diff
@@ -434,6 +449,57 @@
 						url += '?' + queryOptions.join('&')
 					return url
 
+			compileBatch: (data) ->
+				if !utils.isArray(data)
+					throw new Error('Batch must be passed an array')
+				if !mimemessage?.factory
+					throw new Error('The mimemessage implementation must support .factory')
+
+				buildReq = (req) =>
+					# This is a normal request
+					if req.url?
+						contentTransferEncoding = 'binary'
+						if req.body?
+							contentType = 'application/json'
+							b = JSON.stringify(req.body)
+						else
+							contentType = 'application/http'
+							b = ''
+
+						newReq = mimemessage.factory({
+							contentType: contentType
+							contentTransferEncoding: contentTransferEncoding
+							body: "#{req.method} #{@compile(req)} HTTP/1.1\r\n#{b}"
+						})
+
+						for own option, value of req
+							switch option
+								when 'url', 'method', 'body' then continue
+								else newReq.header(option, value)
```

Comment:
`header` isn't declared on the return type of `mimemessage.factory` in the type defintions